staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -506,6 +506,18 @@ func (c *Cacher) Watch(ctx context.Context, key string, opts storage.ListOptions
 	} else {
 		readyGeneration, err = c.ready.waitAndReadGeneration(ctx)
 		if err != nil {
+			if err == ErrStorageInitializing || strings.HasPrefix(err.Error(), "storage is (re)initializing") {
+				// Add audit annotations with OpenShift prefix
+				audit.AddAuditAnnotation(ctx, "openshift.io/watch-reject-reason", "storage_initializing")
```

Comment:
this looks like something that could potentially be added upstream. Have we tried it before and it was rejected?